Sources/Fuzzilli/Fuzzer.swift

```diff
@@ -486,21 +498,46 @@ public class Fuzzer {
 
     /// Process a program that has interesting aspects.
     func processInteresting(_ program: Program, havingAspects aspects: ProgramAspects, origin: ProgramOrigin) {
+        var aspectsToTrack = aspects
+
         func processCommon(_ program: Program) {
             let (newTypeCollectionRun, _) = updateTypeInformation(for: program)
 
             dispatchEvent(events.InterestingProgramFound, data: (program, origin, newTypeCollectionRun))
 
             // All interesting programs are added to the corpus for future mutations and splicing
-            corpus.add(program)
+            corpus.add(program, aspectsToTrack)
+        }
+
+        // If only adding deterministic samples, execute each sample additional times to verify determinism
+        // Each sample will be executed at least minDeterminismExecs, and no more than maxDeterminismExecs times
+        // If two consecutive executions return the same edges after at least minDeterminismExecs times, the sample
+        // is considered deterministic 
+        if deterministicCorpus {
+
+            var foundSameAspects: Bool
+            var newAspects : ProgramAspects = aspects
+            var rounds = 0
```

Comment:
I think this needs to start at `1` since one execution was already performed and, IIUC, the `maxDeterminismExecs` parameters express the total number of executions, not just the additional ones (i.e. with the default values, a sample will be executed at least 3 times and at most 7 times)?